### PR TITLE
Move to Python type annotations & upgrade mypy

### PR DESCRIPTION
**Pull request by gsnedders**
Now we require Python 3.7, we can use type annotations rather than comments. We should take advantage of this.

### CODE REVIEW

File: tools/webdriver/webdriver/error.py [hunk 1]
```diff
@@ -12,8 +12,8 @@ class WebDriverException(Exception):
     # However, http_status need not match, and both are set as instance
     # variables, shadowing the class variables. TODO: Match on both http_status
     # and status_code and let these be class variables only.
-    http_status = None  # type: ClassVar[int]
-    status_code = None  # type: ClassVar[str]
+    http_status: ClassVar[int]
+    status_code: ClassVar[str]
```
Comment (foolip) [inline tools/webdriver/webdriver/error.py:16]: This isn't going to work. I tried the same change but something tries to access `WebDriverException.status_code` and that no longer works.

Comment (gsnedders) [reply tools/webdriver/webdriver/error.py:16]: yikes. well… that seems even worse, because it implies the type is actually wrong.

**Comment by gsnedders** [comment]
FWIW: the mypy upgrade part landed in #39394; we should try and land #39422 which further upgrades mypy soon.

This leaves this PR with `com2ann` plus potentially other coding style changes, given we're gonna introduce plenty of conflicts anyway.

**Comment by jgraham** [comment]
Since #39843 landed I'm going to assume it's OK to close this and we can create a new PR for any remaining changes.

**Pull request closed without merging**